python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -0,0 +1,133 @@
+from typing import Union, Callable, Dict, Iterable, Optional
+
+import numpy as np
+import pandas as pd
+from ray.ml.trainer import GenDataset
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.util import PublicAPI
+from ray.ml.trainer import Trainer
+from ray.ml.checkpoint import Checkpoint
+
+from sklearn.base import BaseEstimator
+from sklearn.model_selection import BaseCrossValidator
+
+# some nice properties of Sklearn models:
+# - they are always pickleable
+# - no logic is present in __init__
+
+ArrayType = Union[pd.DataFrame, np.ndarray]
+MetricType = Union[str, Callable[[BaseEstimator, ArrayType, ArrayType], float]]
+ScoringType = Union[MetricType, Iterable[MetricType], Dict[str, MetricType]]
+CVType = Union[int, Iterable, BaseCrossValidator]
+
+
+@PublicAPI(stability="alpha")
+class SklearnTrainer(Trainer):
+    """A Trainer for scikit-learn estimator training.
+
+    This Trainer runs the ``fit`` method of the given estimator in a
+    non-distributed manner on a single Ray Actor.
```

Comment:
Is the plan to eventually have this use our Ray+joblib integration?

Reply:
Yes, but it is a bit tricky. Not all sklearn estimators use joblib. Some can have only single-node parallelism, and there is no way to detect which uses what. If we set the trainable to use eg. 4 CPUs, then the joblib spawned Actors will not be able to be scheduled. We can perhaps leave it up to the user.